packages/job-worker/src/playout/adlibAction.ts

```diff
@@ -38,81 +38,93 @@ export async function handleExecuteAdlibAction(
 
 		const initCache = await loadPlayoutModelPreInit(context, lock, playlist, false)
 
-		const rundown = initCache.getRundown(playlist.currentPartInfo.rundownId)
-		if (!rundown) throw new Error(`Current Rundown "${playlist.currentPartInfo.rundownId}" could not be found`)
+		return executeAdlibAction(context, playlist, initCache, data)
+	})
+}
 
-		const showStyle = await context.getShowStyleCompound(rundown.showStyleVariantId, rundown.showStyleBaseId)
+export async function executeAdlibAction(
```

Comment:
```suggestion
export async function executeAdlibActionAndSaveModel(
```
This takes the uncommon step of saving the model too, almost everywhere else it the caller that is expected to save the model

Reply:
Updated